examples/routing_service/routing_service_msql_adapter/cpp/ConnectorMsqlDb.cxx

```diff
@@ -0,0 +1,337 @@
+        RTI_RS_ERROR("SQL Error - Allocating the statement handle failed");
+        disconnect();
+        return connected_;
+    }
+    return connected_;
+}
+
+/*
+ * Free up resources and set connected_ to false
+ */
+void ConnectorMsqlDb::disconnect()
+{
+    RTI_RS_LOG_FN(disconnect_db);
+
+    SQLFreeHandle(SQL_HANDLE_STMT, sqlStmtHandle);
+    SQLDisconnect(sqlConnHandle);
+    SQLFreeHandle(SQL_HANDLE_DBC, sqlConnHandle);
+    SQLFreeHandle(SQL_HANDLE_ENV, sqlEnvHandle);
+
+    connected_ = false;
+}
+
+/*
+ * Set connection information and form connection string
+ */
+void ConnectorMsqlDb::setConnectionInfo(const PropertySet &properties)
+{
+    RTI_RS_LOG_FN(setConnectionInfo);
+
+    auto it = properties.find("msql.db.servername");
+    if (it != properties.end()) {
+        server = it->second;
+    } else {
+        RTI_RS_ERROR("Server name not found in properties");
+        return;
+    }
+
+    it = properties.find("msql.db.dbname");
+    if (it != properties.end()) {
+        dbName = it->second;
+    } else {
+        RTI_RS_ERROR("Database name not found in properties");
+        return;
+    }
+
+    it = properties.find("msql.db.tablename");
+    if (it != properties.end()) {
+        tableName = it->second;
+    } else {
+        RTI_RS_ERROR("Table name not found in properties");
+        return;
+    }
+
+    it = properties.find("msql.db.username");
+    if (it != properties.end()) {
+        userName = it->second;
+    } else {
+        RTI_RS_ERROR("Username name not found in properties");
+        return;
+    }
+
+    it = properties.find("msql.db.password");
+    if (it != properties.end()) {
+        password = it->second;
+    } else {
+        RTI_RS_ERROR("Password name not found in properties");
+        return;
+    }
+
+    connectionInfo = "DRIVER={SQL Server}; SERVER=" + server + "; DATABASE="
+            + dbName + "; UID=" + userName + "; PWD=" + password + ";";
+}
+
+/*
+ * Utility method to provide MSQL error information
+ */
+void ConnectorMsqlDb::msqlShowError(
+        unsigned int handletype,
+        const SQLHANDLE &handle)
+{
+    SQLCHAR sqlstate[SQL_BUFF_LEN];
+    SQLCHAR message[SQL_BUFF_LEN];
+
+    if (SQL_SUCCESS
+        == SQLGetDiagRec(
+                handletype,
+                handle,
+                1,
+                sqlstate,
+                NULL,
+                message,
+                SQL_BUFF_LEN,
+                NULL)) {
+        std::cout << "Message: " << message << "\nSQLSTATE: " << sqlstate
+                  << std::endl;
+    }
+}
+
+/*
+ * Write the data to the database
+ * return true on success, otherwise failure
+ */
+auto ConnectorMsqlDb::writeData(const DynamicData *sample) -> bool
+{
+    bool success = true;
+
+    /* write to end position */
+    std::ostringstream cmd(std::ostringstream::ate);
+
+    /* write the data to the database*/
+    cmd << "INSERT INTO " << dbName << ".dbo." << tableName
```

Comment:
This is clearly tied to the `ShapeExtended` data type. That should be documented in the example.

Reply:
Done